### PR TITLE
Update job spec to stop re run in case of failure

### PR DESCRIPTION
**Pull request by husky-parul**
Added method to patch job spec to set active_deadline_seconds 0 in case job failed first time. It will prevent job from running again

### CODE REVIEW

**Comment by husky-parul** [comment]
@danmcp @ravisantoshgudimetla

File: openshift/pman-swift-publisher/put_data.py [hunk 1]
```diff
@@ -58,77 +58,86 @@ def storeData(self, **kwargs):
...
-        swiftStore.storeData(path=os.environ.get('SWIFT_KEY'),out_dir=os.environ.get('OUTGOING_DIR'))
-        shutil.rmtree('/share/outgoing', ignore_errors=True)
-        shutil.rmtree('/share/outgoing/lockfile', ignore_errors=True)
-    else:
-        print('I am not last. Exiting.')
-        exit(0)
+                                # It is possible for the image processing container to error out when that happens, kill the job and exit.
+                                print('Some pod failed so kill job')
+                                # Patch job to terminate. No re run required.
+                                self.patch_job()
```
[the remark this answers is not in the document]
Comment (husky-parul) [reply openshift/pman-swift-publisher/put_data.py]: I can do that. But it is patching job to terminate so named it patch_job

[the remark this answers is not in the document]
Comment (danmcp) [reply openshift/pman-swift-publisher/put_data.py]: What's the method doing?  It doesn't look like a generic patch method right?

[the remark this answers is not in the document]
Comment (husky-parul) [reply openshift/pman-swift-publisher/put_data.py]: Yes, at the moment it only updates the spec to terminate failed job. Updated it to terminate_job

File: openshift/pman-swift-publisher/put_data.py [hunk 2]
```diff
@@ -58,77 +58,86 @@ def storeData(self, **kwargs):
...
-    while True:
-
-        if resrc_version is None:
-            # List all pods for given job using the label_selector to only watch the pods created by the job, 
-            # which is defined by job_id
-            stream = w.stream(v1_client.list_namespaced_pod, namespace, label_selector='job-name='+job_id)
+
+
+
+class KubeClient():
```
Comment (ravisantoshgudimetla) [inline openshift/pman-swift-publisher/put_data.py]: The name could have been better. We have put_data into swift in KubeClient class.

**Pull request merged**